.github/workflows/template-creator.yml

```diff
@@ -0,0 +1,84 @@
+# This workflow should cleanup everything unneeded from the template project
+
+name: Template Creator
+
+on:
+  pull_request:
+  release:
+    types: [published]
+  push:
+    tags:
+    branches:
+      - main
+      - develop
+
+env:
+  TEMPLATES_PATH: ".github/template"
+
+jobs:
+
+  template-create:
```

Comment:
nit: i think it is more like a `template-rename` than a create. (Same goes for the file name)